config/quickwit.yaml

```diff
@@ -80,9 +80,12 @@ version: 0.6
 #     access_key: ${QW_AZURE_STORAGE_ACCESS_KEY}
 #
 #   s3:
+#     access_key_id: ${AWS_ACCESS_KEY_ID}
```

Comment:
We don't want to incentivize users to use the config file to set up their credentials. For instance, this is what you'll find in the documentation of the AWS SDK:

>  It is __NOT__ secure to hardcode credentials into your application, and the credentials
  providers that come with the AWS SDK should be preferred.
  
  I'm saddened we must do so for Azure, but the Azure SDK leaves us no choice. Can you remove `access_key_id` and `secret_access_key`?

Reply:
> We don't want to incentivize users to use the config file to set up their credentials.

That's interesting - from experience this certainly isn't what other applications choose to be doing:

- https://garagehq.deuxfleurs.fr/documentation/connect/apps/
- https://thanos.io/tip/thanos/storage.md/#s3

Hence why this looked a little broken to me at first (and I'm also trying to avoid the craziness of what Thanos has with its configuration of random percentages of options being split between environment variables and YAML configuraiton (on top of which different percentages apply to different components)).

Reply:
I'm ok with leaving those if we add a disclaimer:
```
Hardcoding credentials into configuration files is not secure and strongly discouraged. Prefer the alternative authentication methods that your storage backend may provide.
```

I'll add the same disclaimer to our documentation.